cli/azd/pkg/project/service_target_dotnet_containerapp.go

```diff
@@ -312,6 +234,191 @@ func (at *dotnetContainerAppTarget) Deploy(
 	)
 }
 
+func (at *dotnetContainerAppTarget) deployFromBicep(
+	ctx context.Context,
+	serviceConfig *ServiceConfig,
+	targetResource *environment.TargetResource,
+	remoteImageName string,
+	portNumber int,
+) error {
+	var modulePath string
+
+	appHostRoot := serviceConfig.DotNetContainerApp.AppHostPath
+	if f, err := os.Stat(appHostRoot); err == nil && !f.IsDir() {
+		appHostRoot = filepath.Dir(appHostRoot)
+	}
+
+	synthedPath := filepath.Join(
+		appHostRoot, "infra", fmt.Sprintf("%s.bicep", serviceConfig.DotNetContainerApp.ProjectName))
+	if _, err := os.Stat(synthedPath); err == nil {
+		log.Printf("using bicep module from %s", synthedPath)
+		modulePath = synthedPath
+	} else {
+		log.Printf(
+			"generating container app manifest from %s for project %s",
+			serviceConfig.DotNetContainerApp.AppHostPath,
+			serviceConfig.DotNetContainerApp.ProjectName)
+
+		path, dir, err := apphost.BicepModuleForProject(
+			serviceConfig.DotNetContainerApp.Manifest,
+			serviceConfig.DotNetContainerApp.ProjectName,
+			apphost.AppHostOptions{},
+		)
+		if dir != "" {
+			defer func() { _ = os.RemoveAll(dir) }()
+		}
+		if err != nil {
+			return fmt.Errorf("generating container app manifest: %w", err)
+		}
+		modulePath = path
+	}
+
+	compiledModule, err := at.bicepCli.Build(ctx, modulePath)
+	if err != nil {
+		return fmt.Errorf("compiling bicep module: %w", err)
+	}
+
+	deploymentName := fmt.Sprintf("%s-%d", serviceConfig.DotNetContainerApp.ProjectName, at.clock.Now().Unix())
+
+	// TODO(ellismg): Need to make this better and take the `params` value from the resource into account.
+	params := azure.ArmParameters{
+		"location":                  azure.ArmParameterValue{Value: at.env.Getenv("AZURE_LOCATION")},
+		"containerAppEnvironmentId": azure.ArmParameterValue{Value: at.env.Getenv("AZURE_CONTAINER_APPS_ENVIRONMENT_ID")},
+		"containerRegistryEndpoint": azure.ArmParameterValue{Value: at.env.Getenv("AZURE_CONTAINER_REGISTRY_ENDPOINT")},
+		"managedIdentityClientId":   azure.ArmParameterValue{Value: at.env.Getenv("MANAGED_IDENTITY_CLIENT_ID")},
+		"managedIdentityId":         azure.ArmParameterValue{Value: at.env.Getenv("AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID")},
+		"image":                     azure.ArmParameterValue{Value: remoteImageName},
+		"targetPort":                azure.ArmParameterValue{Value: portNumber},
+	}
+
+	//TODO(ellismg): What tags should we apply to these deployments?
+	res, err := at.deployments.DeployToResourceGroup(
+		ctx,
+		targetResource.SubscriptionId(),
+		targetResource.ResourceGroupName(),
+		deploymentName,
+		json.RawMessage(compiledModule.Compiled), params, nil)
+	if err != nil {
+		return fmt.Errorf("deploying bicep module: %w", err)
+	} else {
+		log.Printf("completed deployment %s for service %s", *res.Name, serviceConfig.DotNetContainerApp.ProjectName)
+	}
+
+	return nil
+}
+
+func (at *dotnetContainerAppTarget) deployFromManifest(
```

Comment:
This method did not functionally change - it was extracted from the `Deploy` method above and put it its own helper method now that we also have one for Bicep.